packages/xrpl/src/client/partialPayment.ts

```diff
@@ -63,7 +63,10 @@ function isPartialPayment(
   }
 
   const delivered = meta.delivered_amount
-  const amount = tx.Amount
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- DeliverMax is a valid field on Payment response
```

Comment:
This doesn't seem like the right way to handle this. We should fix the types.

Reply:
Technically, this is correct because `DeliverMax` is an RPC field and not a protocol field. So that's why it's not added to `Payment` transaction. Refer to this PR for more details - https://github.com/XRPLF/xrpl.js/pull/2689

Reply:
@khancode right, but our library should work around they without type assertions

Reply:
@justinr1234 how do you propose that we use model the `DeliverMax` field?

`DeliverMax` is not a property on the `Payment` transaction type. But V2 rippled responses allude to `DeliverMax` only, not to the erstwhile `Amount` field.

We could use the `any` type for these objects, but that removes all type-information for future engineers. Annotating these objects with `Payment` type (and some ts-lint exceptions) is more informative.